pages/api/auth/[...nextauth].ts

```diff
@@ -99,29 +94,41 @@ export const getAuthOptions = (
 		callbacks: {
 			async jwt({ token, user, account, profile, trigger }: any) {
 				if (trigger === 'signIn') {
-					if (account?.provider == AUTH_PROVIDER.EGO) {
+					const provider = account?.provider;
+					if (provider === AUTH_PROVIDER.EGO) {
 						token.account = account;
 						token.profile = user;
-					} else if (account?.provider == AUTH_PROVIDER.KEYCLOAK) {
+					} else if (provider === AUTH_PROVIDER.KEYCLOAK) {
 						token.account = account;
 						token.profile = profile;
 						token.scopes = await fetchScopes(token.account.access_token);
 					}
+				} else {
+					const tokenExpiresAtMs = token.account.expires_at * 1000;
+					if (Date.now() > tokenExpiresAtMs) {
```

Comment:
just in case the unlikely case of "it expires exactly now" happens...
```suggestion
					if (Date.now() >= tokenExpiresAtMs) {
```

Reply:
added greater or equal to.